SpotGamma.cpp

```diff
@@ -1,249 +1,284 @@
+
+        return;
+    }
 
-SCDLLName("SpotGamma")
-
-const SCString FAKE_URL = "https://spotgamma-system-files.s3.amazonaws.com/ENTER_VALID_URL.csv";
-
-/**************
- * Constants  *
- **************/
+    // Process to initialize variables and pointers along with StateMessage for debug
+    int UpdateIntervalInMin = UpdateInterval.GetInt();
+    int UpdateIntervalInSec = UpdateIntervalInMin * 60;
+    int &LastUpdated = sc.GetPersistentInt(9);
+    int& RequestState = sc.GetPersistentInt(1);
+	enum {REQUEST_NOT_SENT = 0,  REQUEST_SENT, REQUEST_RECEIVED};
+	SCString StateMessage; // Used for debugging purposes.  Should add "if enabled" logic toggle in the future.
+
+    // Only run at time interval
+    // Convert TimeInSec to Seconds and compare to last actual update
+    SCDateTime Now = sc.CurrentSystemDateTime;
+    int TimeInSec = Now.GetTimeInSeconds();
+    
+    if (LastUpdated > 0 && LastUpdated + UpdateIntervalInSec > TimeInSec && sc.Index != 0 && RequestState != REQUEST_SENT) {
+        // exit function if nothing to update
+        return;
+    }
+    LastUpdated = TimeInSec;
 
-enum {REQUEST_NOT_SENT = 0,  REQUEST_SENT};
 
-/*********************
- * Helper Functions  *
- *********************/
+    // hold our HTTP Response
+    // Reference to persistent SCString for storing or accessing HTTP response content
+    SCString &HttpResponseContent = sc.GetPersistentSCString(4);
 
-void cleanDrawings(int lineCountStart, SCStudyInterfaceRef sc) {
-  // clear all drawings
-  int _lineNumber = lineCountStart;
-  while (true) {
-    if (!sc.ChartDrawingExists(sc.ChartNumber, _lineNumber)) {
-      break;
+    SCString Url = UrlPath.GetString();
+    // add specific code to output as CSV attachment from Google Sheets
+    // Reference to how to get Google Sheets to output an editable CSV file: 
+    // https://www.megalytic.com/knowledge/using-google-sheets-to-host-editable-csv-files
+    // Syntax is such: /gvis/tq?tqx=out:csv
+    if (Url == FAKE_URL) {
+        sc.AddMessageToLog("Please input the proper Integration URL", 0);
+        return;
     }
+    Url.Format("%s/gviz/tq?tqx=out:csv", Url.GetChars());
 
-    sc.DeleteACSChartDrawing(sc.ChartNumber, TOOL_DELETE_CHARTDRAWING, _lineNumber);
-    _lineNumber++;
-  }
-}
-
-void drawPriceLevel(
-    SCDateTime beginTime,
-    float price,
-    int lineNumber,
-    int color,
-    int width,
-    SCString label,
-    int fontSize,
-    SCStudyInterfaceRef sc,
-    SubgraphLineStyles lineStyle = LINESTYLE_SOLID)
-{
+    // Cast the pointer retrieved from persistent storage to a pointer to a vector of PriceLabel objects
+    std::vector<PriceLabel>* p_PriceLabels = reinterpret_cast<std::vector<PriceLabel>*>(sc.GetPersistentPointer(0));
+    if (p_PriceLabels == NULL) {
+        p_PriceLabels = new std::vector<PriceLabel>;
+        sc.SetPersistentPointer(0, p_PriceLabels);
+    }
+    else {
+        // clear old levels
+        p_PriceLabels->clear();
+    }
```

Comment:
OK, I think I'm getting confused with this one.  `GetPersistentPointer` returns a `void*&`.  I *think* you want your types to match that.  [Other reading](https://www.geeksforgeeks.org/passing-reference-to-a-pointer-in-c/#:~:text=Below%20program%20shows%20how%20to%20pass%20a%20%E2%80%9CReference%20to%20a%20pointer%E2%80%9D%20to%20a%20function%3A)

I'm used to old-school `void **`, and was expecting `*p_PriceLabels = ...` when setting.  But the reference means that dereferencing happens implicitly.  *However*, the type wasn't initially a reference.

```suggestion
    std::vector<PriceLabel>*& p_PriceLabels = reinterpret_cast<std::vector<PriceLabel>*&>(sc.GetPersistentPointer(PriceLabels));
    if (p_PriceLabels == NULL) {
        p_PriceLabels = new std::vector<PriceLabel>;
        sc.SetPersistentPointer(PriceLabels, p_PriceLabels);
    }
    else {
        // clear old levels
        p_PriceLabels->clear();
    }
```